sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/environment/SapServiceBindingEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,32 @@
+package com.sap.cloud.environment.servicebinding.environment;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.context.properties.bind.BindResult;
+import org.springframework.boot.context.properties.bind.Bindable;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.sap.cloud.environment.servicebinding.SapServiceBindingsProperties;
+
+/* An {@link EnvironmentPostProcessor} that binds service binding properties to {@link SapServiceBindingsProperties}. */
+public class SapServiceBindingEnvironmentPostProcessor implements EnvironmentPostProcessor
+{
+    private static final Logger log = LoggerFactory.getLogger(SapServiceBindingEnvironmentPostProcessor.class);
+
+    @Override
+    public void postProcessEnvironment( ConfigurableEnvironment environment, SpringApplication application )
```

Comment:
(Minor/Comment)

Strange, it looks like `@Nonnull final` is not being enforced here, but it should be.
I thought our pipeline would check for that, I'll investigate...

In the meantime, I'm sorry for sounding pedantic, but we try to declare variables and arguments with `final` wherever possible, as well as declaring them `@Nonnull` or `@Nullable`. The latter also applies to method signatures - for returned objects, excluding void and primitives.

These rules shall only be enforced on `public` API.

Reply:
good hint!